scripts/lib/mkcloud-libvirt.sh

```diff
@@ -57,36 +57,85 @@ function libvirt_net_start()
         ip link set mtu 9000 dev $dev
     done
 
+    boot_mkcloud=/etc/init.d/boot.mkcloud
+    boot_mkcloud_d="$boot_mkcloud.d"
+    boot_mkcloud_d_cloud="$boot_mkcloud_d/$cloud"
+
     if [ -z "$NOSETUPPORTFORWARDING" ] ; then
+        # FIXME: hardcoded assumptions about admin net host range
         nodehostips=$(seq -s ' ' 81 $((80 + $nodenumber)))
-        cat > /etc/init.d/boot.mkcloud <<EOS
+
+        : ${cloud_port_offset:=1100}
+        mkdir -p $boot_mkcloud_d
+        cat > $boot_mkcloud_d_cloud <<EOS
 #!/bin/bash
+# Auto-generated from $0 on `date`
+
+iptables_unique_rule () {
+    # First argument must be chain
+    if iptables -C "\$@" 2>/dev/null; then
+        echo "iptables rule already exists: \$*"
+    else
+        iptables -I "\$@"
+        echo "iptables -I \$*"
```

Comment:
Obviously I put it there deliberately, so I guess you are really asking why I did that. The answer is because it's helpful to see which rules are newly created.